ddl/ddl_api.go

```diff
@@ -223,12 +223,12 @@ func getBatchPendingTiFlashCount(ctx sessionctx.Context) uint32 {
 	return uint32(c)
 }
 
-func (d *ddl) getPendingTiFlashTableCount(sctx sessionctx.Context, originVersion int64) (uint32, int64, bool) {
+func (d *ddl) getPendingTiFlashTableCount(sctx sessionctx.Context, originVersion *int64, pendingCount *uint32) {
 	is := d.infoCache.GetLatest()
 	dbInfos := is.AllSchemas()
 	// If there are no schema change since last time(can be weird)
-	if is.SchemaMetaVersion() == originVersion {
-		return 0, originVersion, false
+	if is.SchemaMetaVersion() == *originVersion {
```

Comment:
Why use pointer? Why not just return the value?

Reply:
I want updating of originVersion inside rather than outside this function, to make code easy to read.

Reply:
Can not it be done like originVersion, pendingCount = d.getPendingTiFlashTableCount()?

Reply:
It should be `originVersion, pendingCount = d.getPendingTiFlashTableCount(originVersion)`
I wrote this originally, and find it very ugly since I have to pass originVersion as an argument and then receive as a result.

Reply:
I think it is ugly now. Whatever, I will leave this to the next reviewer.

Reply:
/cc @zimulala

Reply:
if newVersion > oldVersion { update originCount and originVersion } else { update pendingCount }

Reply:
I don't really understand. The `if else` statement can also be implemented in this function.
Look at this function arguments, `s/update originCount and originVersion/update pendingCount and originVersion`? If it is true, we can do it like "if newVersion > oldVersion { update pendingCount and originVersion } else { update pendingCount and **return  originVersion directly**}"

Reply:
fixed